core/backend/src/ElementTreeWalker.ts

```diff
@@ -447,20 +447,59 @@ export class ElementSubTreeDeleter extends ElementTreeTopDown {
   }
 }
 
+/** Arguments supplied to [[deleteElementTree]].
+ * @beta
+ */
+export interface DeleteElementTreeArgs {
+  /** The iModel containing the elements to delete. */
+  iModel: IModelDb;
+  /** The Id of the root element of the tree to delete. */
+  topElement: Id64String;
+  /** The maximum number of passes to make when deleting definition elements.
+   * Default: 5
+   */
+  maxPasses?: number;
+}
+
 /** Deletes an element tree starting with the specified top element. The top element is also deleted. Uses ElementTreeDeleter.
  * @param iModel The iModel
  * @param topElement The parent of the sub-tree
  * @beta
  */
-export function deleteElementTree(iModel: IModelDb, topElement: Id64String): void {
-  const del = new ElementTreeDeleter(iModel);
-  del.deleteNormalElements(topElement);
-  del.deleteSpecialElements();
+export function deleteElementTree(iModel: IModelDb, topElement: Id64String): void;
+/** Deletes an element tree starting with the specified top element. The top element is also deleted. Uses ElementTreeDeleter.
+ * @param args Specifies the iModel and top element.
+ * @beta
+ */
+export function deleteElementTree(args: DeleteElementTreeArgs): void;
+/** @internal */
+export function deleteElementTree(arg0: DeleteElementTreeArgs | IModelDb, arg1?: Id64String): void {
```

Comment:
Can we keep arg1 as `topElement`?

Reply:
@pmconne ?

Reply:
Sounds a little nitpicky. @nick4598, why not use the "Add a suggestion" to propose your changes?

Reply:
I agree its nitpicky, but I didn't like that I had to read the code to figure out that topElement was still arg1. Maybe that is obvious to people more comfortable with overloads. 

I also didn't use add a suggestion because I figured it would cascade through the function, so I started looking at it locally and it doesn't feel like there is a good way to have arg1 be topElement and still use topElement within the function itself (would have to use a different variable I think within the function). I could be missing something, but it seems bad to change `arg1` for that reason. I will resolve.